src/wallet/api/wallet.cpp

```diff
@@ -1280,6 +1280,42 @@ bool WalletImpl::importOutputs(const string &filename)
     return true;
 }
 
+bool WalletImpl::scanTransactions(const std::vector<std::string> &txids)
+{
+    if (txids.empty())
+    {
+        setStatusError(string(tr("Failed to scan transactions: no transaction ids provided.")));
+        return false;
+    }
+
+    // Parse and dedup args
+    std::unordered_set<crypto::hash> txids_u;
+    for (const auto &s : txids)
+    {
+        crypto::hash txid;
+        if (!epee::string_tools::hex_to_pod(s, txid))
+        {
+            setStatusError(string(tr("Invalid txid specified: ")) + s);
+            return false;
+        }
+        txids_u.insert(txid);
+    }
+    std::vector<crypto::hash> txids_v(txids_u.begin(), txids_u.end());
```

Comment:
You can skip creating two containers (less memory, less copying) for the hashes if you use a `vector` for `txids_u` and call `.reserve()` before the `for` loop.

Reply:
I just copied the simplewallet code. I think unordered set is used to avoid duplicates.

Reply:
Ah, I see the `dedup` part now, is it bad if a TX gets scanned twice?

Reply:
Scanning twice is definitely significantly slower than copying from a set to a vector. I think the current code is fine.

Reply:
That's the simplest way to dedup.